chatkit/src/main/java/com/pusher/chatkit/messages/MessageService.kt

```diff
@@ -45,23 +63,47 @@ class MessageService(
         }
     }
 
+    private val roomListType = object : TypeToken<List<Message>>() {}.type
+
+    fun fetchMessages(): MessageListFutureResult =
+        chatManager.doGet("/rooms/${room.id}/messages")
+            .parseResponseWhenReady()
+
     @JvmOverloads
-    fun sendMessage(message: String, callback: (MessageSentEvent) -> Unit = {}) {
-        currentUser.sendMessage(
-            roomId = room.id,
-            text = message,
-            onCompleteListener = MessageSentListener { id ->
-                callback(MessageSentEvent.Successful(id))
-            },
-            onErrorListener = ErrorListener { error ->
-                callback(MessageSentEvent.Failed(error))
-            }
-        )
+    fun sendMessage(
+        text: String? = null,
+        attachment: GenericAttachment = NoAttachment
+    ): MessageIdResult = when (attachment) {
+        is DataAttachment -> uploadFile(attachment, room.id)
+        is LinkAttachment -> Promise.now(AttachmentBody.Resource(attachment.link, attachment.type).asSuccess<AttachmentBody, elements.Error>())
+        is NoAttachment -> Promise.now(AttachmentBody.None.asSuccess<AttachmentBody, Error>())
```

Comment:
Do you want to create a `Result.promiseSuccess<AttachementBody, Error>(AttachmentBody.None)`? :p

Reply:
I'll put this as a task, as it touches the platform